### PR TITLE
Rating function

### PR DESCRIPTION
**Pull request by khanhkhanh**

### CODE REVIEW

File: db/migrate/20150609074830_create_rating_caches.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+      t.string :dimension
+      t.timestamps
+    end
+    add_index :rating_caches, [:cacheable_id, :cacheable_type]
+  end
+
+  def self.down
+    drop_table :rating_caches
+  end
+end
```
Comment (NguyenTanDuc) [inline db/migrate/20150609074830_create_rating_caches.rb]: new line

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  post '/rate' => 'rater#create', :as => 'rate'
```
Comment (NguyenTanDuc) [inline config/routes.rb]: ' => "

File: db/migrate/20150609074830_create_rating_caches.rb (hunk 1 above)
Comment (vjnhat) [inline db/migrate/20150609074830_create_rating_caches.rb]: add new line

File: app/controllers/rater_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class RaterController < ApplicationController
+  def create
+    if user_signed_in?
+      obj = params[:klass].classify.constantize.find params[:id]
+      obj.rate params[:score].to_f, current_user, params[:dimension]
+      render :json => true
```
Comment (NguyenTanDuc) [inline app/controllers/rater_controller.rb]: coding convention

File: app/controllers/rater_controller.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    else
+      render :json => false
```
Comment (NguyenTanDuc) [inline app/controllers/rater_controller.rb]: coding convention

**Pull request merged**